tsconfig.json

```diff
@@ -1,20 +1,7 @@
 {
-  "compilerOptions": {
-    "target": "es5",
-    "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
-    "skipLibCheck": true,
-    "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true,
-    "strict": true,
-    "forceConsistentCasingInFileNames": true,
-    "noFallthroughCasesInSwitch": true,
-    "module": "esnext",
-    "moduleResolution": "bundler",
-    "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": true,
-    "jsx": "react-jsx"
-  },
-  "include": ["src"]
+  "files": [],
+  "references": [
+    { "path": "./tsconfig.app.json" },
+    { "path": "./tsconfig.node.json" }
+  ]
```

Comment:
**question:** why two tsconfigs? the only one that matters is the one that has `include` for `src` right

like we can just use that tsconfig even for the `vite.config.ts`

Reply:
I think the only significant difference is that the `tsconfig.node.json` file doesn't include types for working with the DOM, so it would be useful to have a separate one for if/when we want to have a backend

Reply:
Would it be possible to factor out the common parts between the two?

Reply:
Sure, let me do that rq

Reply:
Apparent the tsconfig.json is different on https://github.com/chakra-ui/chakra-ui/blob/main/sandbox/vite-ts/tsconfig.json so I'll just use that

Reply:
I reverted the commit, I'm not sure if it was being weird because `vite.config.ts` isn't in `/src/` or if that has something to do with it? I'll just leave it here for now